### PR TITLE
Clean up Doxygen CMake rules

### PR DESCRIPTION
**Pull request by andschwa**
This uses the modern version of the Doxygen CMake package, which
replaces the use of `ExternalProject` and moves the configuration from a
machine-generated file to the `CMakeLists.txt` file.

A slight change to the output of the docs was made: the base directory
no longer uses `include/openenclave` as this is common to tall the
docs (and is implied).

The file `bits/defs.h` was explicitly added because it was being
implicitly pulled in as a dependency (meaning it should have been added
previously).

The `refman` target was renamed to `doxygen`, and the script was
updated. A slight change was made to the script to remove some redundant
code, and fix the output directory.

### CODE REVIEW

**Comment by andschwa** [comment]
This is related to #1014 as the previous implementation caused some issues. @radhikaj Please test this out and let me know.

File: docs/refman/CMakeLists.txt [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
+  set(DOXYGEN_JAVADOC_AUTOBRIEF YES)
+  set(DOXYGEN_QT_AUTOBRIEF YES)
+  set(DOXYGEN_SEPARATE_MEMBER_PAGES YES)
+  set(DOXYGEN_OPTIMIZE_OUTPUT_FOR_C YES)
+  set(DOXYGEN_GENERATE_LATEX NO)
+  set(DOXYGEN_MACRO_EXPANSION YES)
+  set(DOXYGEN_EXPAND_ONLY_PREDEF YES)
+  set(DOXYGEN_PREDEFINED "=OE_ALIGNED(x)=;OE_EXTERNC_BEGIN=")
+
+  doxygen_add_docs(doxygen
```
Comment (radhikaj) [inline docs/refman/CMakeLists.txt]: 👍 
The non default values look good.

**Comment by andschwa** [comment]
@radhikaj I updated the docs 😄

**Pull request merged**